tests/ovs_test.go

```diff
@@ -35,4 +38,52 @@ var _ = Describe("ovs-cni", func() {
 			})
 		})
 	})
+
+	Describe("pod ovs-bridge connectivity tests", func() {
+		const bridgeName = "br-test"
+		Context("When creating ovs-bridge on the host node", func() {
+			BeforeEach( func() {
+				cluster.AddOvsBridgeOnNode(bridgeName)
+			})
+			AfterEach( func() {
+				cluster.RemoveOvsBridgeOnNode(bridgeName)
+			})
+
+			Context("and adding ovs network with vlan", func() {
+				const nadName = "ovs-net"
+				BeforeEach( func() {
+					cluster.CreateNetworkAttachmentDefinition(nadName, bridgeName, `{ "cniVersion": "0.3.1", "type": "ovs", "bridge": "` + bridgeName + `", "vlan": 100 }`)
+				})
+				AfterEach( func() {
+					cluster.RemoveNetworkAttachmentDefinition(nadName)
+				})
+
+				Context("and deploying two pods in the node holding the ovs-bridge", func() {
+					const (
+						pod1Name = "pod-test-1"
+						pod2Name = "pod-test-2"
+						cidrPod1 = "10.0.0.1/24"
+						cidrPod2 = "10.0.0.2/24"
+					)
+					BeforeEach( func() {
+						cluster.CreatePrivilegedPodWithIp(pod1Name, nadName, cidrPod1)
+						cluster.CreatePrivilegedPodWithIp(pod2Name, nadName, cidrPod2)
+					})
+					AfterEach( func() {
+						cluster.DeletePodsInTestNamespace()
+					})
+
+					It("Should enable connectivity between two pods connected to the same bridge", func() {
+						By("Checking pods connectivity by pinging from one to the other")
+						ipPod2, _, err := net.ParseCIDR(cidrPod2)
+						Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("should succeed parsing pod's cidr: %s", cidrPod2))
+
+						out, err := pod.PingFromPod(pod1Name, "test", ipPod2.String())
+						ExpectWithOffset(1, err).ToNot(HaveOccurred(), fmt.Sprintf("should be able to ping from pod %s to IP %s which is on pod %s", pod1Name, ipPod2, pod2Name))
```

Comment:
We don't need WithOffset when we are in a test

Reply:
Done

(but to understand, can you explain why?)

Reply:
You need to use `EventualyWithOffset` when you assert on helpers functions,
so when an error occurs the output will show the line number of the function caused the error on the test code file (instead of the helpers code file).
https://github.com/onsi/gomega/blob/master/gomega_dsl.go#L160

The integer passed to `EventualyWithOffset` is used to calculate the offset in the call stack, `Eventually` warps `EventualyWithOffset` with offset = 0 
https://github.com/onsi/gomega/blob/master/gomega_dsl.go#L206

For example:  you assert on helper function (level 2):
0    Test code (`It`block) 
1             AssertHelper
If an error occurs and you want to return the error line number of the test code level, you need to use `EventualWithOffset(1,..) at `AssertHelper`

As you already know we encourage developers to use assertions in test code only and not in helpers to make the tests easier to understand and maintain :)